upload-api/buckets/invocation-store.js

```diff
@@ -0,0 +1,79 @@
+import {
+  S3Client,
+  PutObjectCommand,
+  ListObjectsV2Command,
+} from '@aws-sdk/client-s3'
+import pRetry from 'p-retry'
+
+/**
+ * Abstraction layer with Factory to perform operations on bucket storing
+ * invocation receipts and indexes.
+ *
+ * @param {string} region
+ * @param {string} bucketName
+ * @param {import('@aws-sdk/client-s3').ServiceInputTypes} [options]
+ */
+export function createInvocationStore(region, bucketName, options = {}) {
+  const s3client = new S3Client({
+    region,
+    ...options,
+  })
+  return useInvocationStore(s3client, bucketName)
+}
+
+/**
+ * @param {S3Client} s3client
+ * @param {string} bucketName
+ * @returns {import('../types').InvocationBucket}
+ */
+export const useInvocationStore = (s3client, bucketName) => {
+  return {
+    /**
+     * Put mapping for where each invocation lives in a Workflow file.
+     *
+     * @param {string} invocationCid
+     * @param {string} workflowCid
+     */
+    putIndex: async (invocationCid, workflowCid) => {
```

Comment:
```suggestion
    putWorkflowIndex: async (invocationCid, workflowCid) => {
```

...might be better to name this "link" rather than "index"